app/src/test/java/org/simple/clinic/user/UserSessionTest.kt

```diff
@@ -416,7 +438,7 @@ class UserSessionTest {
   ) {
     val currentUser = PatientMocker.loggedInUser(pinDigest = "old-digest", loggedInStatus = RESETTING_PIN)
     whenever(userDao.user()).thenReturn(Flowable.just(listOf(currentUser)))
-
+    whenever(facilityRepository.associateUserWithFacilities(any(), any(), any())).thenReturn(Completable.complete())
```

Comment:
Too much mocking to run a test is a sign of a not so good test. Should we add a maintenance task to review the design of the SUT? I guess you are already looking into refactoring this god object `UserSession`.

Reply:
Yup, when I refactor login and registration out of `UserSession`, I'll change the tests to be better.